lib/utils/utils.go

```diff
@@ -439,6 +441,10 @@ func ReadHostUUID(dataDir string) (string, error) {
 func WriteHostUUID(dataDir string, id string) error {
 	err := ioutil.WriteFile(filepath.Join(dataDir, HostUUIDFile), []byte(id), os.ModeExclusive|0400)
 	if err != nil {
+		if errors.Is(err, fs.ErrPermission) {
+			//do not convert to system error as this loses the ability to compare that it is a permission error
```

Comment:
`trace.ConvertSystemError` will convert it to an access denied error, so another option would be to leave this as-is, and update call-site to check `if trace.IsAccessDenied(err) { ...`

Reply:
Leaving as is after discussing with @zmb3 , will likely revisit in the future.